lib/di/interceptor/app_interceptor.dart

```diff
@@ -1,26 +1,84 @@
+import 'dart:io';
+
 import 'package:dio/dio.dart';
+import 'package:flutter_survey/di/di.dart';
 import 'package:flutter_survey/local/shared_preference_helper.dart';
+import 'package:flutter_survey/models/oauth_token.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:flutter_survey/usecase/refresh_token_use_case.dart';
 
-const String HEADER_AUTHORIZATION = 'Authorization';
+const String _headerAuthorization = 'Authorization';
 
-class AppInterceptor extends Interceptor {
-  bool _requireAuthenticate;
-  SharedPreferencesHelper _sharedPreferencesHelper;
+class AppInterceptor extends QueuedInterceptor {
+  final bool _requireAuthenticate;
+  final SharedPreferencesHelper _sharedPreferencesHelper;
+  final Dio _dio;
 
   AppInterceptor(
     this._requireAuthenticate,
     this._sharedPreferencesHelper,
+    this._dio,
   );
 
   @override
-  Future onRequest(
+  void onRequest(
       RequestOptions options, RequestInterceptorHandler handler) async {
     if (_requireAuthenticate) {
-      String accessToken = await _sharedPreferencesHelper.getAccessToken();
-      String tokenType = await _sharedPreferencesHelper.getTokenType();
-      options.headers
-          .putIfAbsent(HEADER_AUTHORIZATION, () => "$tokenType $accessToken");
+      final token = await _buildToken();
+      options.headers.putIfAbsent(_headerAuthorization, () => token);
+    }
+    super.onRequest(options, handler);
+  }
+
+  @override
+  void onError(DioError err, ErrorInterceptorHandler handler) {
+    final statusCode = err.response?.statusCode;
+    if ((statusCode == HttpStatus.forbidden ||
+            statusCode == HttpStatus.unauthorized) &&
+        _requireAuthenticate) {
+      _doRefreshToken(err, handler);
+    } else {
+      handler.next(err);
     }
-    return super.onRequest(options, handler);
+  }
+
+  Future<void> _doRefreshToken(
+    DioError err,
+    ErrorInterceptorHandler handler,
+  ) async {
+    try {
+      final refreshTokenUseCase = getIt<RefreshTokenUseCase>();
+      final result = await refreshTokenUseCase.call();
+      if (result is Success<OAuthToken>) {
+        // Update new token header
+        final newToken = await _buildToken();
+        err.requestOptions.headers[_headerAuthorization] = newToken;
+
+        // Create request with new access token
+        final options = Options(
+            method: err.requestOptions.method,
+            headers: err.requestOptions.headers);
```

Comment:
Can you fix the format code issue here, please? 🙏 

```dart
Options(
    method: err.requestOptions.method,
    headers: err.requestOptions.headers
);

Reply:
@markgravity fixed in https://github.com/luongvo/flutter-survey/pull/78/commits/341ec3a27bf73b076a4a8fdd191a92e3f887b35c